### PR TITLE
fix: migrate from k8s.gcr.io to registry.k8s.io

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Kubernetes is migrating its image registry from k8s.gcr.io to registry.k8s.io.

Part of kubernetes/k8s.io#4780

### CODE REVIEW

**Review by liangyuanpeng — APPROVED** [review]

**Review by sunjayBhatia — COMMENTED** [review]
This is a generated file pulled from https://github.com/kubernetes-sigs/gateway-api so once that releases a new version, this will be updated accordingly

**Comment by ahmedwaleedmalik** [comment]
> This is a generated file pulled from https://github.com/kubernetes-sigs/gateway-api so once that releases a new version, this will be updated accordingly

If I were you, I wouldn't really wait for their release to make this switch. But I'd leave that up to you. As long as you are not affected by https://kubernetes.io/blog/2023/03/10/image-registry-redirect/

**Comment by skriss** [comment]
We can't just update this file statically because it will cause our codegen CI to fail. However, I'm thinking it might be a good idea to add a temporary step to https://github.com/projectcontour/contour/blob/main/Makefile#L253-L259 to replace the old registry with the new. I'm not concerned about it on `main` since we should get an updated Gateway API version before Contour's next release, but it might be good to backport the change to previous releases, just to avoid any issues due to the upcoming redirect. @sunjayBhatia WDYT?

**Comment by sunjayBhatia** [comment]
> We can't just update this file statically because it will cause our codegen CI to fail. However, I'm thinking it might be a good idea to add a temporary step to https://github.com/projectcontour/contour/blob/main/Makefile#L253-L259 to replace the old registry with the new. I'm not concerned about it on `main` since we should get an updated Gateway API version before Contour's next release, but it might be good to backport the change to previous releases, just to avoid any issues due to the upcoming redirect. @sunjayBhatia WDYT?

yeah that sounds like a good idea!

**Comment by skriss** [comment]
@ahmedwaleedmalik I've opened #5175 for this and added you as a co-author, will be closing this one out. Thanks!

**Pull request closed without merging**
